test.c

```diff
@@ -1120,6 +1126,9 @@ static void test_blocking_connection_timeouts(struct config config) {
                   strcmp(c->errstr, "recv timeout") == 0);
 #endif
         freeReplyObject(reply);
+
+        // wait for the DEBUG SLEEP to complete so that Redis server is unblocked for the following tests
+        sleep(3);
     } else {
         test_skipped();
     }
```

Comment:
It could be good to add another testcase after this which covers the found fault.
This would do something like:
```
do_reconnect(c, config); /* Should set a command_timeout */
redisAppendFormattedCommand(c, sleep_cmd, strlen(sleep_cmd));
s = c->funcs->write(c);
..
reply = redisCommand(c, "GET foo");
test_cond(s > 0 && reply == NULL && c->err == REDIS_ERR_IO && strcmp(c->errstr, "Resource temporarily unavailable") == 0);
```
Maybe the `detect_debug_sleep()` could cache its first result to avoid communication problems when/if used in the new testcase.